### PR TITLE
chore(e2e): add `input-date.spec.ts`

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:1]: Deleted cypress tests has caret index checks.

Add it to new playwright tests too.

Use 
```ts
await expect(inputDate.textfield).toHaveJSProperty('selectionStart', 2);
await expect(inputDate.textfield).toHaveJSProperty('selectionEnd', 2);
```

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        });
+
+        ['russian', 'english', 'spanish'].forEach(language => {
+            test(`mobile calendar - ${language}`, async ({page}) => {
+                await tuiGoto(page, '/components/input-date', {language});
+                const api = new TuiDocumentationPagePO(page);
+                const example = api.getExample('#base');
+
+                await example.scrollIntoViewIfNeeded();
+                await api.prepareApiPageBeforeScreenshot(':not(#base)');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:120]: You hide content on not API page)

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                await example.locator('tui-input-date .t-icon tui-svg').click();
+                await expect(page).toHaveScreenshot(`08-input-date-${language}.png`);
+            });
+        });
+
+        test('native data picker', async ({page}) => {
+            await tuiGoto(page, '/components/input-date');
+            const api = new TuiDocumentationPagePO(page);
+            const example = api.getExample('#native-input-date');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:129]:
```suggestion
            const example = new TuiDocumentationPagePO(page).getExample('#native-input-date');
```

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            await expect(page).toHaveScreenshot('04-input-date.png');
+        });
+
+        test.describe('Invalid date cases', () => {
+            test('does not accept day > 31', async ({page}) => {
+                await tuiGoto(page, '/components/input-date/API');
+                await input.focus();
+                await api.prepareApiPageBeforeScreenshot();
+                await page.keyboard.type('35');
+                await expect(page).toHaveScreenshot('05-input-date.png');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:86]: Too large screenshot
05-input-date-chromium-darwin

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            });
+
+            test('does not accept month > 12', async ({page}) => {
+                await tuiGoto(page, '/components/input-date/API');
+                await input.focus();
+                await api.prepareApiPageBeforeScreenshot();
+                await page.keyboard.type('1715');
+                await expect(page).toHaveScreenshot('06-input-date.png');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:94]: Too large screenshot

06-input-date-chromium-darwin

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            });
+
+            test('Type 999999 => 09.09.9999', async ({page}) => {
+                await tuiGoto(page, '/components/input-date/API');
+                await input.focus();
+                await api.prepareApiPageBeforeScreenshot();
+                await page.keyboard.type('999999');
+                await expect(page).toHaveScreenshot('07-input-date.png');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:102]: 07-input-date-chromium-darwin

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            await api.prepareApiPageBeforeScreenshot();
+            await expect(page).toHaveScreenshot('03-input-date.png');
+        });
+
+        test('minimum month more than current month', async ({page}) => {
+            await tuiGoto(page, '/components/input-date/API?min$=3');
+
+            await input.click();
+            await api.prepareApiPageBeforeScreenshot();
+            await expect(page).toHaveScreenshot('04-input-date.png');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:77]: Let's decrease the size of screenshot

04-input-date-chromium-darwin

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+test.describe('InputDate', () => {
+    test.describe('Examples', () => {
+        test('correct filler display for size', async ({page}) => {
+            await tuiGoto(page, '/components/input-date');
+
+            const api = new TuiDocumentationPagePO(page);
+            const example = api.getExample('#sizes');
+
+            await example.scrollIntoViewIfNeeded();
+            await api.prepareApiPageBeforeScreenshot(':not(#sizes)');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:15]: Should we rename this method ?

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:1]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 2 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 3 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 4 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 6 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:79]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 7 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 5 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 1 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+                await input.click();
+                await expect(page).toHaveScreenshot(`01-01-input-date-${size}.png`);
+                await input.clear();
+                await expect(page).toHaveScreenshot(`01-02-input-date-${size}.png`);
+                await input.focus();
+                await page.keyboard.type('01.');
+                await expect(page).toHaveScreenshot(`01-03-input-date-${size}.png`);
+                await input.focus();
+                await page.keyboard.type('06.1994');
+                await expect(page).toHaveScreenshot(`01-04-input-date-${size}.png`);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:33]: 01-02-input-date-s-chromium-darwin

Comment (nsbarsukov) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:33]: You can set different viewport size for different `test.describe()`

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+            await tuiGoto(page, '/components/input-date');
+            const example = new TuiDocumentationPagePO(page).getExample(
+                '#native-input-date',
+            );
+
+            await example.scrollIntoViewIfNeeded();
+            await example.locator('tui-input-date input[type="date"]').fill('2010-03-14');
+            await example
+                .locator('tui-input-date input[type="date"]')
+                .dispatchEvent('change');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:141]: Probably it is cypress legacy. 
I've tried to remove this line and get the same snapshot.

Probably, Playwright already dispatches this action for `<input type="date" />`
https://playwright.dev/docs/input#text-input

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+            const api = new TuiDocumentationPagePO(page);
+            const example = api.getExample('#sizes');
+
+            await example.scrollIntoViewIfNeeded();
+            await api.prepareBeforeScreenshot(':not(#sizes)');
+
+            for (const size of ['s', 'm', 'l']) {
+                const input = example
+                    .locator(`[tuiTextFieldSize="${size}"]`)
+                    .locator('[automation-id="tui-primitive-textfield__native-input"]');
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+    });
+
+    test.describe('API', () => {
+        let api: TuiDocumentationPagePO;
+        let input: Locator;
+
+        test.beforeEach(({page}) => {
+            api = new TuiDocumentationPagePO(page);
+            input = api.apiPageExample.locator(
+                '[automation-id="tui-primitive-textfield__native-input"]',
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+                await expect(input).toHaveJSProperty('selectionEnd', 4);
+                await expect(input).toHaveScreenshot('06-input-date.png');
+            });
+
+            test('Type 999999 => 09.09.9999', async ({page}) => {
+                await tuiGoto(page, '/components/input-date/API');
+                await input.focus();
+                await page.keyboard.type('999999');
+                await expect(input).toHaveJSProperty('selectionStart', 10);
+                await expect(input).toHaveJSProperty('selectionEnd', 10);
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 8 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts:38]: fixed

File: projects/demo-playwright/tests/kit/input-date/input-date.spec.ts (hunk 9 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-date/input-date.spec.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**